### PR TITLE
Add teenytest

### PR DESCRIPTION
**Pull request by pingortle**
This bakes my simple custom runner for TestDouble's [teenytest](https://github.com/testdouble/teenytest) package into vim-test. I followed the basic pattern of existing Ava integration as a template.

This is my first contribution here, and first vimscript PR ever, so I'd appreciate feedback on both form and function.

---

Make sure these boxes are checked before submitting your pull request:

- [x] Add fixtures and spec when implementing or updating a test runner
- [x] Update the README accordingly
- [x] Update the Vim documentation in `doc/test.txt`

### CODE REVIEW

**Comment by pingortle** [comment]
Hmmm… The `:Teenytest` command does not seem to exist with `vim-test` installed from my fork. I was under the impression that it would exist magically since I couldn't find definitions for other commands. I don't personally need that command and only added it to the docs for consistency. I would also happily revert the entry from `doc/test.vim`.

Can anyone provide guidance on this?

**Review by codeinabox — APPROVED** [review]
Thank you for this, looks great!

**Comment by codeinabox** [comment]
> Hmmm… The `:Teenytest` command does not seem to exist with `vim-test` installed from my fork. I was under the impression that it would exist magically since I couldn't find definitions for other commands. I don't personally need that command and only added it to the docs for consistency. I would also happily revert the entry from `doc/test.vim`.
> 
> Can anyone provide guidance on this?

For the command to be available, you need to [add it to the runner commands configuration](https://github.com/vim-test/vim-test#runner-commands)

File: spec/fixtures/teenytest/test/normal.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  'Adds two numbers': function () {
+    // assertions
+  }
+}
```
Comment (codeinabox) [inline spec/fixtures/teenytest/test/normal.js:5]: Could you update this with an example from their docs, this is probably a good one to go with: https://github.com/testdouble/teenytest/blob/main/example/simple-node/test/lib/dog-test.js

**Review by codeinabox — CHANGES_REQUESTED** [review]

File: spec/fixtures/teenytest/test/normal.js (hunk 1 above)
Comment (pingortle) [reply spec/fixtures/teenytest/test/normal.js]: For sure! Something like this https://github.com/vim-test/vim-test/pull/705/commits/c329ec5570ade6810175b3976dedfe97b7f3f115?

**Review by codeinabox — APPROVED** [review]

**Pull request merged**